docs/notebook/Tutorial.ipynb

```diff
@@ -618,20 +618,52 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "### Environment variable interpolation\n",
-    "\n",
-    "Environment variable interpolation is also supported.\n",
-    "\n",
-    "Input yaml file:"
+    "Interpolations may be nested, enabling more advanced behavior like dynamically selecting a sub-config:"
```

Comment:
I reverted the changes to the notebook in d7edd05
Taking note of creating a new PR updating the notebook once this one is merged.

Reply:
Ah actually some changes to the notebook are required for tests to pass. I restored it for now, I'll check later how to make it easier to review.

Reply:
I found a way to diff the notebook, check the diff here: https://odelalleau.github.io/diff_omegaconf_notebook.html

Note that it includes a minor fix I just pushed in 1aaf32c

Reply:
> Use dedent here (or alternatively, use a dictionary for the input instead of a string):

I'd rather not use a dictionary since it's less readable for this example. I'll switch it to dedent, but what's the motivation? (using dedent makes it more verbose and I don't see any benefit here)

Reply:
So, I made most recommended changes in c9488b0
You can see the diff vs the previous version at https://odelalleau.github.io/diff_omegaconf_notebook_v2.html

Comments:

1. I'll backport the same changes to `usage.rst` as well, unless you have any objection
2. I kept `pop()` instead of `del` to avoid KeyErrors. Alternatively, I could just remove that line since it's unlikely for `DB_PASSWORD` to be in the env variables (it used to be an issue before when we were setting it). Keeping it for now just to be safe.
3. Not convinced we need a `print_x()` to print the type, since it's clear from the output (that's why I'm printing the `repr()` of the objects). Happy to add it if you really want it.

Reply:
> 1. I'll backport the same changes to `usage.rst` as well, unless you have any objection

Done in 6895659 (the only thing I didn't change is the grouping of statements in the env example showing various data types, because in the doc we can have one output per line so grouping doesn't actually make it more readable)